pos-v2/contracts/Protocol.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity 0.5.16;
+
+contract Protocol {
+
+    event ProtocolVersionChanged(string deploymentSubset, uint256 protocolVersion, uint256 asOfBlock);
+
+    struct DeploymentSubset {
+        uint version;
+        uint asOfBlock;
+    }
+
+    mapping (string => DeploymentSubset) deploymentSubsets;
+    address governor;
+
+    modifier onlyGovernor() {
+        require(msg.sender == governor, "caller is not the protocol contract governor");
+
+        _;
+    }
+
+    constructor(address _governor) public {
+        governor = _governor;
+    }
+
+    function deploymentSubsetExists(string calldata deploymentSubset) external view returns (bool) {
+        return deploymentSubsets[deploymentSubset].version > 0;
+    }
+
+    function setProtocolVersion(string calldata deploymentSubset, uint256 protocolVersion, uint256 asOfBlock) external onlyGovernor {
+        if (deploymentSubsets[deploymentSubset].version == 0) {
+            require(asOfBlock == 0, "initial protocol version must be from block 0");
```

Comment:
calling this method with `0,0` doesn't make sense. consider disallowing this variation and separate to a dedicated method call